protocol-specs/generators/block_processing_delegate/index.js

```diff
@@ -0,0 +1,210 @@
+		missedBlocks: 1,
+		producedBlocks: 0,
+		rank: 70,
+		fees: 0,
+		rewards: 0,
+		vote: 9999899990000000,
+		productivity: 0,
+	},
+];
+
+// Object holding the genesis account information and passphrase as well as
+// an existing delegate account for DEVNET
+// TODO: Move this to devnet.json config file.
+const accounts = {
+	// Genesis account, initially holding 100M total supply
+	genesis: {
+		address: '16313739661670634666L',
+		publicKey:
+			'c094ebee7ec0c50ebee32918655e089f6e1a604b83bcaa760293c61e0f18ab6f',
+		passphrase:
+			'wagon stock borrow episode laundry kitten salute link globe zero feed marble',
+		balance: '10000000000000000',
+		encryptedPassphrase:
+			'iterations=1&salt=e8c7dae4c893e458e0ebb8bff9a36d84&cipherText=c0fab123d83c386ffacef9a171b6e0e0e9d913e58b7972df8e5ef358afbc65f99c9a2b6fe7716f708166ed72f59f007d2f96a91f48f0428dd51d7c9962e0c6a5fc27ca0722038f1f2cf16333&iv=1a2206e426c714091b7e48f6&tag=3a9d9f9f9a92c9a58296b8df64820c15&version=1',
+		password: 'elephant tree paris dragon chair galaxy',
+	},
+	existingDelegate: {
+		address: '10881167371402274308L',
+		publicKey:
+			'addb0e15a44b0fdc6ff291be28d8c98f5551d0cd9218d749e30ddb87c6e31ca9',
+		passphrase:
+			'actress route auction pudding shiver crater forum liquid blouse imitate seven front',
+		balance: '0',
+		delegateName: 'genesis_100',
+	},
+	futureDelegate: {
+		passphrase:
+			'blame address tube insect cost knock major level regret bring april stick',
+		privateKey:
+			'b92e223981770c716ee54192a0ad028639d28d41221b72e455447bc4767aeb94caff2242b740a733daa3f3f96fc1592303b60c1704a8ac626e2704da039f41ee',
+		publicKey:
+			'caff2242b740a733daa3f3f96fc1592303b60c1704a8ac626e2704da039f41ee',
+		address: '2222471382442610527L',
+		balance: '0',
+	},
+};
+
+const generateTestCasesValidBlockDelegateRegistration = () => {
+	const chainStateBuilder = new ChainStateBuilder(
+		genesisBlock,
+		initialAccountsState,
+		accounts,
+	);
+
+	chainStateBuilder
+		.transfer('50')
+		.from('16313739661670634666L')
+		.to('10881167371402274308L')
+		.forge();
+
+	chainStateBuilder
+		.transfer('30')
+		.from('10881167371402274308L')
+		.to('2222471382442610527L')
+		.forge();
+
+	chainStateBuilder
+		.registerDelegate('RadioHead')
+		.for('2222471382442610527L')
+		.forge();
+
+	const chainAndAccountStates = chainStateBuilder.getScenario();
+
+	return {
+		initialState: {
+			chain: chainAndAccountStates.chain.slice(0, 2),
+			accounts: chainAndAccountStates.initialAccountsState,
+		},
+		input: chainAndAccountStates.chain.slice(2),
+		output: {
+			chain: chainAndAccountStates.chain,
+			accounts: chainAndAccountStates.finalAccountsState.slice(-1),
+		},
+	};
+};
+
+const generateTestCasesInvalidBlockDelegateRegistrationSecondTime = () => {
+	const chainStateBuilder = new ChainStateBuilder(
+		genesisBlock,
+		initialAccountsState,
+		accounts,
+	);
+
+	chainStateBuilder
+		.transfer('50')
+		.from('16313739661670634666L')
+		.to('10881167371402274308L')
+		.forge();
```

Comment:
`to` maybe only relevant to transfer transaction or maybe there are other usecase in mind?
to be easily combine multiple transactions, make it make sense to do like
```
chainStateBuilder
  .transfer('16313739661670634666L', '50', '10881167371402274308L')
  .registerSecondSignature('16313739661670634666L', 'passphrase')
  .forge()
```

Reply:
This is opinionated but I feel that the way Pablo implemented it feels more natural. If doing it the way you propose, you need to know the order of the parameters. Also reading `transfer` `from` `to` resembles the way you say it in plain English.

Reply:
The idea is to experiment how this will work. One of the points is what Dani says but also with the current implementation we don't need any validation or schema as there's only one way to call the methods. I would like for us to give it a try to this to check how it feels.